src/ReactDOMRe.re

```diff
@@ -41,6 +41,41 @@ let renderToElementWithId = (reactElement, id) =>
   | Some(element) => render(reactElement, element)
   };
 
+module Unstable = {
+  type root;
+
+  [@bs.module "react-dom"]
+  external createRoot: Dom.element => root = "unstable_createRoot";
+
+  [@bs.send] external render: (root, React.element) => unit = "";
+
+  let createRootWithClassName = className =>
+    switch (_getElementsByClassName(className)) {
+    | [||] =>
+      raise(
+        Invalid_argument(
+          "ReactDOMRe.Unstable.createRootWithClassName: no element of class "
+          ++ className
+          ++ " found in the HTML.",
+        ),
+      )
+    | elements => createRoot(Array.unsafe_get(elements, 0))
+    };
+
+  let createRootWithId = id =>
+    switch (_getElementById(id)) {
+    | None =>
+      raise(
+        Invalid_argument(
+          "ReactDOMRe.Unstable.createRootWithId: no element of id "
+          ++ id
+          ++ " found in the HTML.",
+        ),
+      )
```

Comment:
We just swapped to logging here instead of raising to avoid the additional dependency - could you move to that pattern here as well?

Reply:
The other functions return `unit`. What do you suggest we return in the `None` case for this branch, given that the `Some` branch is returning a `root`?

Reply:
My suggestion is that the `createRootWith*` family of functions return `result(root, string)` type. That way we don't need to log, and can return something that the application can handle.